### DISCUSSION TITLE
Why are there a lot of makefiles in the language stats of my repository？

### DISCUSSION THREAD

**Discussion by NaroZeol** [community/community#122143] [Repositories] [answered]
### Select Topic Area

Question

### Body

Here is my [repo](https://github.com/NaroZeol/CSAPPLAB)

I am sure that it has no makefile right now. But the language stats say I have 88% makefile.
![image](https://github.com/community/community/assets/113886086/a06f8c6b-ef7d-4833-9363-ddbb86054d6e)

Anyone know why and how to solve it?

**Comment by charlestanev** [comment]
The presence of multiple Makefiles in the language stats of the repository could be due to various reasons:

Multiple Projects: The repository might contain multiple projects or subdirectories, each with its own Makefile for building and compiling the project.
Different Environments: Each Makefile could be tailored for different environments or configurations within the project.
Historical Reasons: Over time, as the project evolved or different contributors worked on it, multiple Makefiles might have been added or modified to serve different purposes.
To address this issue and consolidate the Makefiles:

Review and Merge: Review the functionality and contents of each Makefile to identify commonalities and differences. Merge similar Makefiles into a single, well-organized Makefile that can handle all required tasks.
Refactor and Modularize: Refactor the Makefile(s) to make them more modular and maintainable. Consider breaking down complex tasks into smaller, reusable components and organizing them logically.
Standardization: Standardize the structure and naming conventions across the Makefile(s) to ensure consistency and clarity. Document the purpose and usage of each target and variable within the Makefile(s).
Version Control: Ensure that changes to the Makefile(s) are properly version-controlled and documented. Use descriptive commit messages to explain the rationale behind each change.
Testing and Validation: Test the consolidated Makefile thoroughly to verify that it performs all required tasks correctly and efficiently. Validate its functionality across different environments and scenarios to ensure compatibility.
By consolidating and optimizing the Makefile(s), you can simplify the build process, improve maintainability, and enhance the overall development experience within the repository.

**Reply by markolalovic** [reply]
This reply is very likely AI generated, see [gptzero](https://gptzero.me/) results:

![gptzero results](https://i.ibb.co/Y0v3qVG/gptzero-results.png)

Do not just post AI-generated content verbatim, see [Code of Conduct](https://github.com/orgs/community/discussions/69006) guidelines around using generative AI in Community Discussions.

**Comment by markolalovic** [comment, marked as answer]
Hi, your repository has some files with .d extension in directory: `attacklab/解题过程文件`. 

You can see this from the breakdown of files by language of your repository [CSAPPLAB](https://github.com/NaroZeol/CSAPPLAB):
```bash
$ github-linguist --breakdown
87.99%  1086832    Makefile
8.62%   106461     C
3.38%   41721      Assembly
0.02%   225        Shell

Makefile:
attacklab/解题过程文件/Part1/Level2/injectCode.d
attacklab/解题过程文件/Part1/Level3/injectCode.d
attacklab/解题过程文件/Part2/level2/rtarget.d
attacklab/解题过程文件/Part2/level3/gadgetTest/decrease.d
attacklab/解题过程文件/Part2/level3/gadgetTest/increase.d
attacklab/解题过程文件/Part2/level3/gadgetTest/push.d
```

Note that:
> The . d file extension is also commonly used for dependency files generated by GCC. These files are small makefiles listing dependencies of a source file that get included into the main makefile, causing make to know to rebuild a source file when one of the included header files was changed.

See also this discussion: [Github thinks this file is Makefile instead of D #3462](https://github.com/github-linguist/linguist/issues/3462).

**Reply by NaroZeol** [reply]
Thanks.
It worked after I remove them.
👍
